sos/upload/targets/redhat.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2024 Red Hat, Inc. Jose Castillo <jcastillo@redhat.com>
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import os
+import json
+from sos.upload.targets import UploadTarget
+from sos.utilities import convert_bytes, TIMEOUT_DEFAULT
+from sos.policies.auth import DeviceAuthorizationClass
+from sos.policies.distros.redhat import RHELPolicy
+from sos import _sos as _
+
+try:
+    import requests
+    REQUESTS_LOADED = True
+except ImportError:
+    REQUESTS_LOADED = False
+
+
+class RHELUpload(UploadTarget):
```

Comment:
Maybe worth renaming this to `RHELUploadTarget` to align with `UploadTarget` class name logic? (and ditto `UbuntuUpload`)

Reply:
makes sense, let me do that

Reply:
Actually, I'm thinking that here we will be perhaps too much? We'll call this "target RHELUpload" or "RHELUpload target" but adding target to the name itself may be unnecessary? But I'm not a native speaker so even if it sounds weird to me, it may be absolutely right to change it

Reply:
Well the `RHELUploadTarget` seems ugly long to me, but imho the class serves target functionality more than upload functionality. So more likely `RHELTarget` than `RHELUploadTarget`..?

Whatever works for others, esp. native speakers :)